### PR TITLE
Fix Crossgen2 extra arguments escape hatch

### PR DESCRIPTION
**Pull request by davidwrighton**
The current implementation of the extra arguments escape hatch does not allow multiple arguments to be passed to crossgen2. That is necessary for the upcoming profile guided optimization work. In the long term, we will add specific project properties/itemgroups for pgo, but for now, this will unblock the work, and be useful going forward.

### CODE REVIEW

**Comment by AntonLapounov** [comment]
Should we fix the command line parser instead of doing this change?

File: src/Tasks/Microsoft.NET.Build.Tasks/RunReadyToRunCompiler.cs
Comment (AntonLapounov) [inline src/Tasks/Microsoft.NET.Build.Tasks/RunReadyToRunCompiler.cs:221]: If you pass `StringSplitOptions.RemoveEmptyEntries`, you may skip checking for the empty string.

File: src/Tasks/Microsoft.NET.Build.Tasks/RunReadyToRunCompiler.cs
Comment (AntonLapounov) [inline src/Tasks/Microsoft.NET.Build.Tasks/RunReadyToRunCompiler.cs:219]: Is it possible to use the space as a separator? Note that at present the command line parser does not allow `--option value` format in response files: you have to separate the value with either `:` or `=`.

File: src/Tasks/Microsoft.NET.Build.Tasks/RunReadyToRunCompiler.cs
Comment (trylek) [inline src/Tasks/Microsoft.NET.Build.Tasks/RunReadyToRunCompiler.cs:239]: Could we move the for loop after the then-else conditional blocks to remove the duplication now that the code has become a bit more complex?

**Review by trylek — APPROVED** [review]
LGTM, thank you!

File: src/Tasks/Microsoft.NET.Build.Tasks/RunReadyToRunCompiler.cs
Comment (davidwrighton) [reply src/Tasks/Microsoft.NET.Build.Tasks/RunReadyToRunCompiler.cs:219]: No, we can't use space. It is reasonable for arguments to contain spaces. For instance, one of the cases I specifically want to work with involves passing file names through this mechanism, which on Windows means we need to support spaces.

File: src/Tasks/Microsoft.NET.Build.Tasks/RunReadyToRunCompiler.cs
Comment (AntonLapounov) [inline src/Tasks/Microsoft.NET.Build.Tasks/RunReadyToRunCompiler.cs:216]: I think this check is no longer required.

File: src/Tasks/Microsoft.NET.Build.Tasks/RunReadyToRunCompiler.cs
Comment (AntonLapounov) [reply src/Tasks/Microsoft.NET.Build.Tasks/RunReadyToRunCompiler.cs]: I see. The semicolon is also valid in file names, though definitely less frequently used. Note that at present SDK allows separating extra arguments with `%0a`, e.g., `-p:PublishReadyToRunCrossgen2ExtraArgs="--inputbubble%0a--composite"`.

**Comment by AntonLapounov** [comment]
The syntax is
```C#
Crossgen2ExtraCommandLineArgs.Split(new[] { ';' }, StringSplitOptions.RemoveEmptyEntries)
```
You might want to use `StringSplitOptions.TrimEntries` as well, otherwise `--foo; --bar` would not work.

**Comment by davidwrighton** [comment]
@dsplaisted Can you review/merge this if its good? I don't have permissions to this repo, and I believe you are more an expert here than I am.

**Comment by dsplaisted** [comment]
@davidwrighton This code is owned by @AntonLapounov, and it's generally his or his team's responsibility to approve / merge PRs to that code.

**Comment by AntonLapounov** [comment]
@davidwrighton I think you have the same permissions as I do.  The checks must be green in order to merge this PR, otherwise the merge button is disabled.

**Pull request merged**
